### PR TITLE
414 resolve light warnings for mylyn bugzilla connector

### PR DESCRIPTION
**Pull request by gnl42**

### CODE REVIEW

File: mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.core/src/org/eclipse/mylyn/internal/bugzilla/core/XmlCleaner.java [hunk 1]
```diff
@@ -58,8 +58,6 @@ public static BufferedReader clean(Reader in, File tempFile) {
 					content.append(token.toString());
 				}
 			}
-			content.flush();
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.core/src/org/eclipse/mylyn/internal/bugzilla/core/XmlCleaner.java:61]: I would keep this statement

File: mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.tests/src/org/eclipse/mylyn/bugzilla/tests/core/BugzillaConfigurationTest.java [hunk 2]
```diff
@@ -112,19 +114,12 @@ public void testNullCacheFile() {
...
-		try {
-			File tempFile = File.createTempFile("XmlCleaner-", "tmp");
-			tempFile.deleteOnExit();
-
-			InputStream stream = BugzillaFixture.getResource("testdata/configuration/rdfconfig218.txt");
-			BufferedReader in = new BufferedReader(new InputStreamReader(stream));
-			try (in) {
-				inCleaned = XmlCleaner.clean(in, tempFile);
-				if (tempFile != null) {
-					tempFile.delete();
```
Comment (ruspl-afed) [inline mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.tests/src/org/eclipse/mylyn/bugzilla/tests/core/BugzillaConfigurationTest.java:125]: I'm not sure why this statement was needed right here, but in refactored version it is moved to another location for some reason.

**Review by akurtakov — APPROVED** [review]

**Review by ruspl-afed — COMMENTED** [review]
@akurtakov was faster, but please consider my comments @gnl42

File: mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.core/src/org/eclipse/mylyn/internal/bugzilla/core/XmlCleaner.java (hunk 1 above)
Comment (gnl42) [reply mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.core/src/org/eclipse/mylyn/internal/bugzilla/core/XmlCleaner.java:61]: close() does a flush

File: mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.tests/src/org/eclipse/mylyn/bugzilla/tests/core/BugzillaConfigurationTest.java (hunk 2 above)
Comment (gnl42) [reply mylyn.tasks/connectors/bugzilla/org.eclipse.mylyn.bugzilla.tests/src/org/eclipse/mylyn/bugzilla/tests/core/BugzillaConfigurationTest.java:125]: It made more sense to move the delete() to the final block

**Pull request merged**